src/hci/gnomex/utility/WorkItemSolexaPrepParser.java

```diff
@@ -90,8 +90,14 @@ private void initializeSample(Element n, Sample sample) throws Exception {
       sample.setSeqPrepLibConcentration(new BigDecimal(n.getAttributeValue("seqPrepLibConcentration")));
     } else {
       sample.setSeqPrepLibConcentration(null);
-    }    
-    
+    }
+
+    if (n.getAttributeValue("libPrepPerformedByID") != null && !n.getAttributeValue("libPrepPerformedByID").equals("")) {
+      sample.setLibPrepPerformedByID(new Integer(n.getAttributeValue("libPrepPerformedByID")));
+    } else {
+      sample.setLibPrepPerformedByID(null);
+    }
+
```

Comment:
General refactoring note with implications for CORE development - we have code that repeats this type of logic many places and it could be cleaned up nicely with some good utility methods in common code.